src/algebra/continued_fractions/definitions/basic.lean

```diff
@@ -0,0 +1,321 @@
+⟨λ ⟨a, b⟩, ⟨(a : β), (b : β)⟩⟩
+
+@[simp, move_cast]
+lemma coe_to_generalized_continued_fraction_pair {a b : α} :
+  (↑(gcf.pair.mk a b) : gcf.pair β) = gcf.pair.mk (a : β) (b : β) :=
+rfl
+
+end coe
+end generalized_continued_fraction.pair
+
+/--
+A generalised continued fraction consists of a head term `h` and a sequence of
+generalized_continued_fraction.pairs `s`.
+-/
+structure generalized_continued_fraction :=
+(h : α) (s : seq $ generalized_continued_fraction.pair α)
+
+variable {α}
+
+namespace generalized_continued_fraction
+open generalized_continued_fraction as gcf
+
+/-- Returns the sequence of partial numerators `aᵢ` of `g`. -/
+def partial_numerators (g : gcf α) : seq α := g.s.map gcf.pair.a
+/-- Returns the sequence of partial denominators `bᵢ` of `g`. -/
+def partial_denominators (g : gcf α) : seq α := g.s.map gcf.pair.b
+
+/-- A gcf terminates at position `n` if its sequence terminates at position `n`. -/
+def terminated_at (g : gcf α) (n : ℕ) : Prop := g.s.terminated_at n
+
+/-- It is decidable whether a gcf terminates at a given position. -/
+instance terminated_at_decidable (g : gcf α) (n : ℕ) : decidable (g.terminated_at n) :=
+by { unfold terminated_at, apply_instance }
+
+/-- A gcf terminates if its sequence terminates. -/
+def terminates (g : gcf α) : Prop := g.s.terminates
+
+/- Interlude: define some expected coercions. -/
+section coe
+/-- Fix another type `β` and assume `α` can be converted to `β`. -/
+variables {β : Type*} [has_coe α β]
+
+/-- Coerce a sequence by elementwise coercion. -/
+def seq.coe_to_seq : has_coe (seq α) (seq β) := ⟨seq.map (λ a, (a : β))⟩
+
+local attribute [instance] seq.coe_to_seq
+
+/-- Coerce a gcf by elementwise coercion. -/
+instance has_coe_to_generalized_continued_fraction : has_coe (gcf α) (gcf β) :=
+⟨λ ⟨h, s⟩, ⟨(h : β), (s : seq $ gcf.pair β)⟩⟩
+
+@[simp, move_cast]
+lemma coe_to_generalized_continued_fraction {g : gcf α} :
+(↑(g : gcf α) : gcf β) = ⟨(g.h : β), (g.s : seq $ gcf.pair β)⟩ :=
+by { cases g, refl }
+
+end coe
+end generalized_continued_fraction
+
+/-
+We now continue to define simple continued fractions. A *simple continued fraction* (scf) is a gcf
+whose partial numerators are all equal to one.
+
+                                1
+                h + --------------------------
+                                  1
+                      b₀ + --------------------
+                                    1
+                            b₁ + --------------
+                                        1
+                                  b₂ + --------
+                                      b₃ + ...
+
+
+Again, for convenience, one writes `[h; b₀, b₁, b₂,...]`.
+-/
+
+/--
+A generalized continued fraction is a simple continued fraction (scf) if all partial numerators are
+equal to one.
+-/
+def generalized_continued_fraction.is_simple_continued_fraction
+(g : generalized_continued_fraction α) [has_one α] :
```

Comment:
This line should be indented, and merged with the next line

Reply:
Fixed.